docs/diagnostics/PX1033.md

```diff
@@ -0,0 +1,81 @@
+# PX1033
+This document describes the PX1033 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                 | Type  | Code Fix    | 
+| ------ | ------------------------------------------------- | ----- | ----------- | 
+| PX1033 | The DAC does not have an explicit primary key declaration | Warning (ISV Level 3: Informational) | Available | 
+
+## Diagnostic Description
+You can define a primary key field in a DAC. The diagnostic suggests adding a primary key definition to a DAC. 
+The warning is shown only for DACs that have the `PXCacheName` or `PXPrimaryGraph` attribute.
+The diagnostic does not check abstract DACs and DACs which have only unbound DAC properties (fully-unbound DACs used for custom popups and filters for inquiry forms).
+
+To fix the issue, add the PK class to the DAC.
+
+The code fix generates a primary key and the 'Find' method which allows to easily find DAC row by providing values composing primary key.
+
+## Example of Code for Which the Warning Is Shown
+
+```C#
+[PXCacheName(Messages.SOOrder)]
+public class SOOrder : IBqlTable
+{
+	#region OrderType
+	public abstract class orderType : PX.Data.BQL.BqlString.Field<orderType> { }
+	protected String _OrderType;
+	[PXDBString(2, IsKey = true, IsFixed = true, InputMask = ">aa")]
+	...
+	public virtual String OrderType
```

Comment:
The example is a bit trivial, we can use more close to real life example by adding OrderNbr - this way you will have a real SOOrder key. 
Also, it is very outdated style - currently everyone uses auto-properties in C# instead of explicit backing field, so the example could be rewritten like this:
```csharp
[PXCacheName(Messages.SOOrder)]
public class SOOrder : IBqlTable
{
	#region OrderType
	public abstract class orderType : PX.Data.BQL.BqlString.Field<orderType> { }

	[PXDBString(2, IsKey = true, IsFixed = true, InputMask = ">aa")]
	...
	public virtual String OrderType {get; set; }
	#endregion
        
        #region OrderNbr
	[PXDBString(IsKey = true, InputMask = "")]
	[PXDefault]
	[PXUIField(DisplayName = "Order Nbr.")]
	public string OrderNbr { get; set; }

	public abstract class orderNbr : IBqlField { }
	#endregion
}
```